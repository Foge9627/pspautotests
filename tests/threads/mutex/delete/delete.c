#include "../sub_shared.h"

SETUP_SCHED_TEST;

#define DELETE_TEST(title, mutex) { \
	int result = sceKernelDeleteMutex(mutex); \
	if (result == 0) { \
		printf("%s: OK\n", title); \
	} else { \
		printf("%s: Failed (%X)\n", title, result); \
	} \
}

int main(int argc, char **argv) {
	DELETE_TEST("NULL", 0);
	DELETE_TEST("Invalid", 0xDEADBEEF);

	SceUID mutex = sceKernelCreateMutex("delete", 0, 0, NULL);
	DELETE_TEST("Valid", mutex);
	DELETE_TEST("Twice", mutex);

	BASIC_SCHED_TEST(
		sceKernelDeleteMutex(mutex2);
	);

	return 0;
}