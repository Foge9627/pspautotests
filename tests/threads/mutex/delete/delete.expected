NULL: Failed (800201C3)
Invalid: Failed (800201C3)
Valid: OK
Twice: Failed (800201C3)
Scheduling: ABCDE
